app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataPresenter.kt

```diff
@@ -34,7 +37,17 @@ class MetadataPresenter @Inject constructor(
             return
         }
 
-        view.fillForm(ViewStateModel(sample.disease, sample.images.toList(), emptyList()))
+        val lastMetadata = repository.last()?.metadata
+        view.fillForm(
+            ViewStateModel(
+                sample.disease,
+                sample.images.toList(),
+                emptyList(),
+                smearType = lastMetadata?.let { MetadataMapper.getSmearTypeId(it.smearType) },
```

Comment:
This is correct as it is, but just as a personal preference, even if we have a singleton/object, I like to hide this kind of implementation detail by actually injecting the instance of it. Just something for you to consider if you want, I am not strongly holding to it.

Reply:
Can you review changeset c0f5c9692349070bbd81db509eab0ed21928085e for this change? I wanted to confirm that you meant that I should pass in the instance of the SampleMetadata to the getSmearTypeId, etc. functions.

Reply:
I was referring to the instance of `MetadataMapper` :P Sorry for not being clear. I can create a small commit to show you what I meant.

Reply:
https://github.com/eduardb/GDD-app/commit/02844d765e137fbc7271ec96eb4c53e959677feb
Feel free to use the same approach.

Reply:
Thanks for the code assist! I'm adding it exactly, except reverting back to passing in the values (smearType, species, stage) instead of the entire SampleMetadata.